src/providers/postgres/CMakeLists.txt

```diff
@@ -56,8 +56,14 @@ INCLUDE_DIRECTORIES(
   ${CMAKE_CURRENT_BINARY_DIR}/../../ui
 )
 
+ADD_LIBRARY (postgresprovider_a STATIC ${PG_SRCS} ${PG_HDRS} ${PG_MOC_SRCS})
 ADD_LIBRARY (postgresprovider MODULE ${PG_SRCS} ${PG_HDRS} ${PG_MOC_SRCS})
 
+TARGET_LINK_LIBRARIES (postgresprovider_a
+  ${POSTGRES_LIBRARY}
+  qgis_core
+  qgis_gui
```

Comment:
I've tried to remove it and it goes kaboom when compiling qgspgsourceselect.cpp.
Looks like the GUI for configuring the provider is in the provider. Make sense, to fix that we have to split every providers in two modules.